### ISSUE TITLE
Throw errors in hooks?

### ISSUE THREAD

**Issue by ryanfields** [matb33/meteor-collection-hooks#145]
Is there any way to throw an error in a hook such that the original collection method's callback will get the error?  I'm hoping to perform validations in before.insert and before.update.  At the moment, I can't find any way to pass validation errors up to the callback of the original insert/update call.  All I can seem to do is return false in the hook, which doesn't give the caller any info about why the method failed.

**Comment by voidale** [comment]
@ryanfields found a solution for this one? +1

**Comment by joelpickup** [comment]
@ryanfields @voidale struggling with this as well... did either of you manage this?

**Comment by matb33** [comment]
Thanks to @aramk's pull request, I've finally implemented this feature in 0.7.15

**Comment by tagrudev** [comment]
So what's the correct way to use it in a hook so that Collection.update catches the error thrown ?

I tried something like this

```coffeescript
# nothing happens
MyCollection.before.update (userId, doc) ->
  new Meteor.Error(400, "The user must be logged in to post a comment")
  return false

MyCollection.before.update (userId, doc) ->
  throw new Meteor.Error(400, "The user must be logged in to post a comment")
```
